### ISSUE TITLE
Multiple axes with different colormaps do not work.

### ISSUE THREAD

**Issue by EraYaN** [matlab2tikz/matlab2tikz#915]
Matlab version: 2016a
OS: Win 10 x64
m2t version: 1.0.0

We have 2 axes that are overlayed.

Both have a different colormaps and images made using `imagesc` and are overlayed.

For example

```matlab
n = 500;
data1 = peaks(n);
data2 = flip(peaks(n));
ax1 = axes;
ax2 = axes;
linkaxes([ax1,ax2]);
xlim([1 n]);
ylim([1 n]);

axes(ax1);
img1 = imagesc(data1);
set(img1 ,'AlphaData',0.5*ones(n));
colormap(ax1,'hot'); % the ax1, argument breaks everything

view(2);
axes(ax2);
img2 = imagesc(data2);
set(img2 ,'AlphaData',0.5*ones(n));
colormap(ax2,'bone'); % the ax2, argument breaks everything
c2 = colorbar(ax2);

ax2.Visible = 'off';
ax2.XTick = [];
ax2.YTick = [];

set([ax1,ax2],'Position',[.10 .11 .685 .815]);

% update visual
drawnow();

matlab2tikz('filename','out.tikz','height', '\figureheight', 'width', '\figurewidth');
```

this exports out.tikz and out-1.png and out-2.png, but both use the default colormap. (The blue and yellow stuff.)

**Comment by miscco** [comment]
@EraYaN did you try the development version?

**Comment by miscco** [comment]
Hm i have trouble understanding, what you actually want. This seems like a really special case. Can you post what you intend (Export the matlab figure directly as a jpeg). How do you overlay the maps

**Comment by miscco** [comment]
So quick analysis gives me some headaches. The problem is, that the colorbar is a `figure` property. That said, it is currently impossible to assign multiple colorbars directly.

So what you have to do seems to be somehow complicated but achievable.

[1.] Create the individual files via 
```matlab
img1 = imagesc(data1);
set(img1 ,'AlphaData',0.5*ones(n));
colormap('hot');
matlab2tikz('Data_a.tex', 'standalone', true);

close
img2 = imagesc(data2);
set(img2 ,'AlphaData',0.5*ones(n));
colormap('bone');
matlab2tikz('Data_b.tex', 'standalone', true);
```

[2.] Create your original file via the code you have posted, except that you add a 
```matlab
set(gcf, 'Colormap', bone);
```

[3.] In the resulting tex file you change the Names of the png to Data_a-1.png and Data_b-1.png respectively

For me that recreated your original matlab figure. I am sorry, that there is no easier way, but that is what matlab gives us.

**Comment by EraYaN** [comment]
Option 2 gives both images in that colormap. (not intended)

This is the actual result I want: https://erayan.com/shared/MATLAB_2016-06-10_13-40-27.png

These are 1 base image (the 'hot') one and 4 overlayed images ('bone').

It exports fine (with alpha channels and all) it just "sources" the colors wrong. It should take the axes colors not the figure colors.

And for option 1, how do I get latex to typeset these onto each other? (without manually editing the TikZ?) I could always just run the script twice with two different colors and "merge" the results, it's juts not nice.

**Comment by miscco** [comment]
@EraYaN That werent different options but steps you should do to get it working

**Comment by EraYaN** [comment]
Aah yes I see now, that would require editing the TikZ, I'm working with a great many figures, so I used this (for now, until m2t supports these kinds of constructions):

```matlab
 set(gcf, 'Colormap', col);
        matlab2tikz('filename',strcat(mfilename(),'/quantization-noise.tikz'),'standalone',false,'height', '\figureheight', 'width', '\figurewidth');
        set(gcf, 'Colormap', col2);
        matlab2tikz('filename',strcat(mfilename(),'/quantization-noise-bone.tikz'),'standalone',false,'height', '\figureheight', 'width', '\figurewidth');
        movefile(strcat(mfilename(),'/quantization-noise-bone-2.png'),strcat(mfilename(),'/quantization-noise-2.png'),'f');
        movefile(strcat(mfilename(),'/quantization-noise-bone-3.png'),strcat(mfilename(),'/quantization-noise-3.png'),'f');
        movefile(strcat(mfilename(),'/quantization-noise-bone-4.png'),strcat(mfilename(),'/quantization-noise-4.png'),'f');
        movefile(strcat(mfilename(),'/quantization-noise-bone-5.png'),strcat(mfilename(),'/quantization-noise-5.png'),'f');
        delete(strcat(mfilename(),'/quantization-noise-bone.tikz'));
        delete(strcat(mfilename(),'/quantization-noise-bone-1.png'));
```

Which is hella ugly but it works. And I can run this for the about 60 scripts we have.

**Comment by miscco** [comment]
Yeah, that should be fine. You can either move the pngs or edit the tikz file. Actually i like you way more.

However, from the MATLAB functionality i would guess that we will not be able to support this at all. Again the problem is the identification of the whole figure with a unique colormap.

**Comment by egeerardyn** [comment]
@miscco `colorbar` is indeed a figure property in HG1. In HG2, however, I thought that one of the changes was to allow multiple colormaps. So we should be able to support it.
And if I also remember correctly, there are some tricks to make HG1 support multiple colormaps as well (but since HG1 is slowly being deprecated, I would not waste any time on those).

**Comment by miscco** [comment]
Hm in Matlab2015a colomap is still a figure property. It obviously works somhow by using colormap(ax, ...) but i cannot find the information anywhere.

**Comment by egeerardyn** [comment]
I agree with you that it is not an axis property (not directly, or as documented).

However, I had a little play with `uiinspect` and it seems to be stored in `gca.ColorSpace.Colormap` (the easiest way is to cast each level to a struct and then inspect it). Anyhow, it is quite easy to access the colormap programmatically without relying on undocumented behavior: `colormap(gca)`.
And that behavior is backwards compatible to HG1 (R2014a at least).

The possible problem that I see is that in some cases, we need to associate a colormap with a colorbar. I suppose this can be done, but I haven't checked this.
